src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -3252,6 +3252,12 @@ private string GetUriPartsFromUserString(UriComponents uriParts)
             }
         }
 
+        // Cut trailing spaces
+        private void GetLengthWithoutTrailingSpaces(string str, ref ushort length ,int idx)
+        {                        
+            while (length > idx && UriHelper.IsLWS(str[length - 1])) --length;                         
```

Comment:
This repeated mutation of the length passed by ref could have a measurable perf impact.

Reply:
Just out of curiosity, why does it matter (performance wise) that length is passed by ref?

Reply:
> why does it matter (performance wise) that length is passed by ref?

When it's passed by value, the compiler can enregister the value and then all of the decrements of the value are just manipulating that register.  When it's passed by ref, the compiler doesn't know if/when the target of the ref changes, so it needs to dereference the pointer each time the value is decremented.

e.g. this:
```C#
using System;
using System.Diagnostics;
using System.Runtime.CompilerServices;

class Program
{
    static void Main()
    {
        const int Decs = 1_000_000_000;
        var sw = new Stopwatch();
        while (true)
        {
            int value;

            sw.Restart();
            value = int.MaxValue;
            value = MutateLocal(Decs, value);
            sw.Stop();
            Console.WriteLine("MutateLocal: " + sw.Elapsed.TotalSeconds);

            sw.Restart();
            value = int.MaxValue;
            MutateRef(Decs, ref value);
            sw.Stop();
            Console.WriteLine("MutateRef:   " + sw.Elapsed.TotalSeconds);

            Console.WriteLine();
        }
    }

    [MethodImpl(MethodImplOptions.NoInlining)]
    static int MutateLocal(int n, int value)
    {
        for (int i = 0; i < n; i++) value--;
        return value;
    }

    [MethodImpl(MethodImplOptions.NoInlining)]
    static void MutateRef(int n, ref int value)
    {
        for (int i = 0; i < n; i++) value--;
    }
}
```
produces this on my machine:
```
MutateLocal: 0.3207588
MutateRef:   1.6596342

MutateLocal: 0.3122484
MutateRef:   1.6347019

MutateLocal: 0.3136665
MutateRef:   1.6396611
```

Reply:
Yes, that's fine.

Reply:
thank's for precious advice!